### PR TITLE
adding signal filters

### LINKED ISSUE #26: Add filters for signals
**Is your feature request related to a problem? Please describe.**
We need to add a filtering capability so that a user can apply the filter the signal payload.

**Describe the solution you'd like**
JQ Filter looks like a good candidate for this. 
https://stedolan.github.io/jq/manual/
https://github.com/savaki/jq

### PR DESCRIPTION
**Pull request by magaldima**
closes #26

### CODE REVIEW

File: examples/time-filter-webhook.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    sensors.argoproj.io/controller-instanceid: axis
+spec:
+  signals:
+    - name: webhook
+      webhook:
+        endpoint: /app
+        method: POST
+      filters:
+        time:
+          start: "2016-05-10T15:04:05Z07:00"
```
Comment (shrinandj) [inline examples/time-filter-webhook.yaml]: Does this mean that the trigger will be fired only if the signal is fired between the `start` and `stop` time?

File: pkg/apis/sensor/v1alpha1/types.go [hunk 2]
```diff
@@ -165,18 +165,54 @@ type ResourceSignal struct {
...
+const (
+	JSONTypeBool   JSONType = "bool"
+	JSONTypeNumber JSONType = "number"
+	JSONTypeString JSONType = "string"
+)
+
+// DataFilter describes constraints and filters for event data
+// Regular Expressions are purposefully not a feature as they are overkill for our uses here
+// See Rob Pike's Post: https://commandcenter.blogspot.com/2011/08/regular-expressions-in-lexing-and.html
+type DataFilter struct {
```
Comment (shrinandj) [inline pkg/apis/sensor/v1alpha1/types.go]: Can you add an example for DataFilter as well?

Comment (shrinandj) [reply pkg/apis/sensor/v1alpha1/types.go]: Never mind.. Just saw it.

File: controller/signal-filter_test.go [hunk 3]
```diff
@@ -0,0 +1,316 @@
+package controller
```
Comment (shrinandj) [inline controller/signal-filter_test.go]: Great job on adding unit tests! :-)

**Comment by magaldima** [comment]
These generated conflicts are annoying.. are these a feature of the 
```
[[constraint]]
  name = "k8s.io/code-generator"
  branch = "release-1.10"
```
dependency?

Anyway, do you know if it matters how I resolve these conflicts?

**Comment by magaldima** [comment]
@shrinandj rebased on top of master, can you re-approve?

**Pull request merged**